executor/write.go

```diff
@@ -180,10 +181,11 @@ func updateRecord(ctx sessionctx.Context, h int64, oldData, newData []types.Datu
 
 // resetErrDataTooLong reset ErrDataTooLong error msg.
 // types.ErrDataTooLong is produced in types.ProduceStrWithSpecifiedTp, there is no column info in there,
-// so we reset the error msg here, and wrap old err with errors.Wrap.
+// so we reset the error msg here, and wrap old err with new error.
 func resetErrDataTooLong(colName string, rowIdx int, err error) error {
 	newErr := types.ErrDataTooLong.Gen("Data too long for column '%v' at row %v", colName, rowIdx)
-	return errors.Wrap(err, newErr)
+	log.Error(err)
```

Comment:
remove the log?

Reply:
I think this is a problem with `ErrDataTooLong` not being a struct with an `error` field. But we should create the chain, and we can do it this way (this could be moved upstream to pingcap/errors):

``` go
type ErrorChain struct {
  newErr: error
  oldErr: error
}
func (e ErrorChain) Error() string { return newErr.Error() }
func (e ErrorChain) Unwrap() error { return oldErr }
```

Reply:
https://github.com/pingcap/errors/pull/6

Reply:
You could try uber-go/multierr.